ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py

```diff
@@ -11892,3 +11893,270 @@ def test_torch_tensor_quantile(
         method_flags=method_flags,
         on_device=on_device,
     )
+
+
+@st.composite
+def put_along_axis_helper(draw):
+    input_dtype, x, axis, shape = draw(
+        helpers.dtype_values_axis(
+            available_dtypes=["int64"],
```

Comment:
I don't think we should be hardcoding this to `int64`. Same for the `test_tensor_functions` and the experimental api helper.

Reply:
It only works with `int64` because it casts this dtype to the rest of the arguments in method input dtypes. The index tensor has to be of type `int64`.

Reply:
You can pass lists of dtypes to `test_frontend_method`. `init_input_dtypes` would be this `input_dtype` which can be any of supported types, `method_input_dtypes` would be [`int64`, int64`, `input_dtype`] (if values has to be the same dtype as `x`, otherwise would be another `val_dtype` which `put_along_axis_helper` would have to return as well).

Reply:
Also for the test for `ivy.put_along_axis` the test is failing unless the available dtypes in the helper is set to `["int64"]`, probably caused by the same issue here.

The tests for the scatter instance methods because of this error 
```
ivy.utils.exceptions.IvyBackendException: torch: put_along_axis: Index tensor must have the same number of dimensions as self tensor
```
with this set of examples:
```
args=('int64', array([[0, 0],  [0, 0]]), array([[0, 0]]), array([[0, 0]]), 0)
```

These are the values being produced by the strategy. They run fine when you test them yourself with the function in a script somewhere but they are failing the testing pipeline somehow.

I also set the max value inside the index tensor to be either 0 or 1, because I couldn't find a way to draw only values that would return an index error when performing the slicing operations. The mechanism works fine though, its only that sometimes it could draw values which were greater than the length of `arr[indices]` in  the `_handle_reduction_op` helper function.

Reply:
Hi @mobley-trent, I tried what I suggested it to you and `test_torch_scatter` passes for all backends except paddle. Please do that to solve the `int64` issue.

Reply:
Does `init_input_dtypes = [input_dtype]` and `method_input_dtypes = ["int64", input_dtype]` work for the methods?

Reply:
Basically `init_input_dtypes` should provide the dtypes for all arrays contained in `init_all_as_kwargs_np`, and likewise `method_input_dtypes` should provide the dtypes for all arrays contained in `method_all_as_kwargs_np`.